### PR TITLE
CORE-6759- update vnode api docs

### PR DESCRIPTION
**Pull request by driessamyn**
NOTE:
Build is failing on Open API test as I need to re-generate this, but it's a moving target at the moment, so will do that after the tech writers review.

### CODE REVIEW

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt [hunk 1]
```diff
@@ -12,7 +12,7 @@ import net.corda.libs.virtualnode.endpoints.v1.types.VirtualNodeInfo
 /** RPC operations for virtual node management. */
 @HttpRpcResource(
     name = "Virtual Node API",
-    description = "Virtual node management endpoints.",
+    description = "The Virtual Nodes API consists of a number of endpoints to manage Virtual Nodes.",
```
Comment (nadinequinr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt]:
```suggestion
    description = "The Virtual Nodes API consists of a number of endpoints to manage virtual nodes.",
```

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt [hunk 2]
```diff
@@ -39,9 +39,9 @@ interface VirtualNodeRPCOps : RpcOps {
      * @throws `HttpApiException` If the request returns an exceptional response.
      */
     @HttpRpcGET(
-        title = "List all virtual nodes in the cluster",
-        description = "List all virtual nodes in the cluster.",
-        responseDescription = "List details of the all virtual nodes in the cluster."
+        title = "List all virtual nodes",
```
Comment (nadinequinr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt]:
```suggestion
        title = "Lists all virtual nodes",
```

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt [hunk 3]
```diff
@@ -7,25 +7,29 @@ import net.corda.libs.cpiupload.endpoints.v1.CpiIdentifier
  *
  * These field names are what the end-users see.
  */
+/**
+ * Virtual Node Info
+ *
+ * @param holdingIdentity The holding identity of this virtual node.
```
Comment (nadinequinr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt:13]: Can we elaborate on this at all? It doesn't really give any more info than the name.

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt [hunk 4]
```diff
@@ -7,25 +7,29 @@ import net.corda.libs.cpiupload.endpoints.v1.CpiIdentifier
...
+ * @param cpiIdentifier The CPI that this virtual node uses.
+ * @param vaultDdlConnectionId Vault DDL DB connection ID.
+ * @param vaultDmlConnectionId Vault DML DB connection ID.
+ * @param cryptoDdlConnectionId Crypto DDL DB connection ID.
+ * @param cryptoDmlConnectionId Crypto DML DB connection ID.
+ * @param uniquenessDdlConnectionId Uniqueness DDL DB connection ID.
+ * @param uniquenessDmlConnectionId Uniqueness DML DB connection ID.
+ * @param hsmConnectionId HSM connection ID.
+ * @param state The virtual node state.
```
Comment (nadinequinr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt]:
```suggestion
 * @param state The state of the virtual node.
```

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt [hunk 5]
```diff
@@ -7,25 +7,29 @@ import net.corda.libs.cpiupload.endpoints.v1.CpiIdentifier
...
+ */
```
Comment (nadinequinr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt:23]: As above, all of these descriptions are a bit terse.

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodes.kt [hunk 6]
```diff
@@ -1,9 +1,14 @@
 package net.corda.libs.virtualnode.endpoints.v1.types
 
 /**
- * The data object received via HTTP in response to a request to get virtual nodes in the cluster.
+ * This class is serialized and returned as JSON in the REST api
```
Comment (nadinequinr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodes.kt]:
```suggestion
 * This class is serialized and returned as JSON in the REST API.
```

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodes.kt [hunk 7]
```diff
@@ -1,9 +1,14 @@
...
  *
- * @param virtualNodes List of virtual nodes.
+ * These field names are what the end-users see.
+ */
+/**
+ * List of Virtual Nodes
+ *
+ * @param virtualNodes List of virtual node info.
```
Comment (nadinequinr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodes.kt]: Should this link to VirtualNodeInfo?

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt (hunk 3 above)
Comment (driessamyn) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt:13]: Holding Identity is a fundamental concept of Corda 5 (similar to Virtual Node or Application Network), so I'm not sure how much we can or should elaborate without a quite detailed explanation.

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt (hunk 5 above)
Comment (driessamyn) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodeInfo.kt:23]: I fixed the wording a bit (just order), but I don't think we can explain much more. Most of these fields aren't actually very useful for the users at the moment, but will be in future when we support bring your own DB and possibly for node upgrade. They can also be useful in a debugging scenario. So we may refer to these fields in other documentation or troubleshooting guides, but I don't think we need to add more details descriptions.

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt [hunk 8]
```diff
@@ -12,7 +12,7 @@ import net.corda.libs.virtualnode.endpoints.v1.types.VirtualNodeInfo
 /** RPC operations for virtual node management. */
 @HttpRpcResource(
     name = "Virtual Node API",
-    description = "Virtual node management endpoints.",
+    description = "The Virtual Nodes API consists of a number of endpoints to manage virtual nodes.",
```
Comment (paulmoloneyr3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt:15]: I would add a short description of what a virtual node is. From here:

https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4059365421/2022+07+07+-+Virtual+Nodes+-+Blog+Post

It's quite long, perhaps a summary is;


"A virtual node represents a Corda identity, a person or business that wants to interact with other people or businesses using Corda. The virtual node also contains everything needed to communicate and transact on Corda: keys, certificates, and storage. Corda virtualizes the execution of flow process steps, allowing flows for multiple identities and from multiple applications to be executed within the same environment at the same time. An instance of a virtual node is created for long enough to handle the execution steps needed and then allowed to dematerialize."

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt (hunk 8 above)
Comment (driessamyn) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt:15]: I disagree. If somebody doesn't know what a Virtual Node is, they won't be using this API. I think it just adds noise.

The above description is also not technically correct. We can't ever create "an instance of a virtual node". It's not a thing, no instance can ever exist. It's a context in which something is executed.

Comment (driessamyn) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt:15]: Maybe we need a separate data dictionary. Similar to: https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/tweet

It's not content that belongs in the code though.

Comment (paulmoloneyr3) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt:15]: Hi Dries, do you know where I can find a more accurate description of virtual nodes? That was the most recently updated description I could find (and the first hit on Confluence), cheers.

Comment (driessamyn) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt:15]: https://github.com/corda/platform-eng-design/blob/master/core/corda-5/corda-5.0/virtual-node/onboarding-journey.md#definition-of-terms is probably the best we have. At least that I am aware of.

Comment (nadinequinr3) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt:15]: https://nadine.preview.docs.r3.com/en/platform/corda/5.0-dev-preview-2/introduction/key-concepts.html#virtual-nodes

(This was reviewed by Kat)

**Review by nadinequinr3 — DISMISSED** [review]

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt (hunk 8 above)
Comment (driessamyn) [reply libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/VirtualNodeRPCOps.kt:15]: Maybe that's conceptually how we want to refer to them, but it's not technically how it works. We create instances of Sandboxes, which have the application code (CorDapp) and execute in the Virtual Node context.

For example, there is no `VirtualNode` type in the code. There is a `VirtualNodeInfo` class that has metadata about a virtual node, and we have a `VirtualNodeContext` type that is used to define the context of a Sandbox.

This API is about managing the Virtual Node metadata (Identity, Vault connection, keys etc).

**Review by paulmoloneyr3 — DISMISSED** [review]

File: libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodes.kt [hunk 9]
```diff
@@ -1,9 +1,14 @@
...
 
 /**
- * The data object received via HTTP in response to a request to get virtual nodes in the cluster.
+ * This class is serialized and returned as JSON in the REST API.
  *
- * @param virtualNodes List of virtual nodes.
+ * These field names are what the end-users see.
+ */
+/**
+ * List of Virtual Nodes
```
Comment (AnnaK-R3) [inline libs/virtual-node/virtual-node-endpoints/src/main/kotlin/net/corda/libs/virtualnode/endpoints/v1/types/VirtualNodes.kt]:
```suggestion
 * List of virtual nodes
```

File: processors/rpc-processor/src/integrationTest/kotlin/net/corda/processors/rpc/OpenApiCompatibilityTest.kt [hunk 10]
```diff
@@ -83,9 +84,16 @@ class OpenApiCompatibilityTest {
...
+            prefix = "open-api-baseline", suffix = ".json")
+        File(tmpBaselineFile.toUri()).printWriter().use {
+            it.println(existingSwaggerJson.second.toJson())
+        }
+
         assertThat(diffReport).withFailMessage(
             "Produced Open API content:\n" + existingSwaggerJson.first +
-                    "\nis different to the baseline. Differences noted: ${diffReport.joinToString(" ## ")}"
+                    "\nis different to the baseline. Differences noted: ${diffReport.joinToString(" ## ")}\n\n" +
+                    "New baseline written to: $tmpBaselineFile"
```
Comment (driessamyn) [inline processors/rpc-processor/src/integrationTest/kotlin/net/corda/processors/rpc/OpenApiCompatibilityTest.kt:96]: @conalsmith-r3 , I added this so it is easier to get the new baseline. Are you ok with this? If so, can you approve on behalf of the HTTP-RPC team?

**Review by AnnaK-R3 — APPROVED** [review]

**Review by conal-smith — APPROVED** [review]
LGTM

**Pull request merged**